src/core/device.c

```diff
@@ -114,12 +160,128 @@ static void device_done(Unit *u) {
         assert(d);
 
         device_unset_sysfs(d);
+        free(d->devname);
         d->wants_property = strv_free(d->wants_property);
 }
 
+static int root_cgroup_apply_block_device_io_cost_qos(Unit *root_slice_unit, Device *d) {
+        char buf[DECIMAL_STR_MAX(dev_t)*2+9+(6+DECIMAL_STR_MAX(uint64_t)+1)*10+(6+DECIMAL_STR_MAX(loadavg_t)+1)*2];
+        IOCostQos *io_cost_qos;
+        dev_t dev;
+        int r;
+
+        assert(root_slice_unit);
+        assert(d);
+
+        io_cost_qos = &d->io_cost_qos;
```

Comment:
Is there a real benefit to having a separate variable for this? We don't really save too much on typing. I'd either go further and abbreviate the variable `icq` to actually save on typing and reading or remove it altogether. same for `io_cost_model` further down.

Reply:
This is mostly for readability. I would find `icq` to be less readable.